pypeit/core/wavecal/patterns.py

```diff
@@ -585,6 +585,112 @@ def curved_quadrangles(detlines, linelist, npixels, detsrch=5, lstsrch=10, pixto
     return dindex[1:, :], lindex[1:, :], wvcen[1:], disps[1:]
 
 
+def solve_xcorr(detlines, linelist, dindex, lindex, line_cc, patt_dict=None, nreid_min = 4, cc_local_thresh = 0.8):
+    """  Given a starting solution, find the best match for all detlines
+
+    Parameters
+    ----------
+    detlines : ndarray
+      list of detected lines in pixels (sorted, increasing)
+    linelist : ndarray
+      list of lines that should be detected (sorted, increasing)
+    dindex : ndarray
+      Index array of all detlines (pixels) used in each triangle
+    lindex : ndarray
+      Index array of the assigned line (wavelengths)to each index in dindex
+    patt_dict : dict
+      Contains all relevant details of the fit
```

Comment:
Please describe the various keys here.

Reply:
Changed the logic here so that pattern dict is a return value, and added docs.